### ISSUE TITLE
Pursue/complete exising work to add JSON output to smbstatus

### ISSUE THREAD

**Issue by phlogistonjohn** [samba-in-kubernetes/samba-container#46]
As part of the metrics story started in #41 we think the cleanest approach is to get the metrics from smbstatus as JSON. There's existing (unmerged) work to add JSON output to samba commands. If we were able to help complete this work for smbstatus it can serve as the basis for our metric collection.

**Comment by raghavendra-talur** [comment]
Link to the merge request on samba-team gitlab repo that adds json output to smbstatus. 
https://gitlab.com/samba-team/samba/-/merge_requests/1269

**Comment by anoopcs9** [comment]
> Another [MR](https://gitlab.com/samba-team/samba/-/merge_requests/2449)(currently draft) coming along the way...

@synarete Here is a sample output:
```
# smbstatus --json | jq
{
  "sessions": [
    {
      "pid": "0:79204",
      "uid": 0,
      "gid": 0,
      "username": "root",
      "groupname": "root",
      "remote_machine": "192.168.122.1",
      "hostname": "ipv4:192.168.122.1:46078",
      "session_dialect": "SMB3_11",
      "encryption": "-",
      "signing": "partial(AES-128-GMAC)"
    }
  ],
  "shares": [
    {
      "service": "gluster-vol",
      "pid": "0:79204",
      "machine": "192.168.122.1",
      "connected_at": "Tue Mar 29 05:26:07 PM 2022 IST",
      "encryption": "-",
      "signing": "-"
    }
  ],
  "locked_files": [
    {
      "pid": "0:79204",
      "uid": 0,
      "denymode": "DENY_NONE",
      "access_mask": {
        "hex": "0x00000003"
      },
      "oplock": {
        "EXCLUSIVE": false,
        "BATCH": false,
        "LEVEL_II": false,
        "LEASE": false
      },
      "lease": {},
      "service_path": "/",
      "filename": "MAINTAINERS"
    }
  ]
}


# smbstatus

Samba version 4.17.0pre1-GIT-d7a91a855c7
PID     Username     Group        Machine                                   Protocol Version  Encryption           Signing              
----------------------------------------------------------------------------------------------------------------------------------------
0:79204 root         root         192.168.122.1 (ipv4:192.168.122.1:46078)  SMB3_11           -                    partial(AES-128-GMAC)

Service      pid     Machine       Connected at                     Encryption   Signing     
---------------------------------------------------------------------------------------------
gluster-vol  0:79204 192.168.122.1 Tue Mar 29 05:26:07 PM 2022 IST  -            -           


Locked files:
Pid          User(ID)   DenyMode   Access      R/W        Oplock           SharePath   Name   Time
--------------------------------------------------------------------------------------------------
0:79204      0          DENY_NONE  0x3         RDWR       NONE             /   MAINTAINERS   Tue Mar 29 17:26:12 2022

```
MR has been marked _Ready_ for review.

**Comment by synarete** [comment]
@anoopcs9 This is so great! many thanks, it makes the life of smb metrics-exporter much more simple and less error-prone.

**Comment by anoopcs9** [comment]
An updated version:
```
# smbstatus --json | jq
{
  "timestamp": "2022-04-04T19:09:17.645718+0530",
  "version": "4.17.0pre1-GIT-4171736339b",
  "smb.conf": "/usr/local/etc/samba/smb.conf",
  "sessions": {
    "1:20614": {
      "pid": "1:20614",
      "uid": 0,
      "gid": 0,
      "username": "root",
      "groupname": "root",
      "remote_machine": "192.168.122.1",
      "hostname": "ipv4:192.168.122.1:40536",
      "session_dialect": "SMB3_11",
      "encryption": {
        "cipher": "",
        "degree": "none"
      },
      "signing": {
        "cipher": "AES-128-GMAC",
        "degree": "partial"
      }
    }
  },
  "shares": {
    "1:20614": {
      "service": "gluster-vol",
      "pid": "1:20614",
      "machine": "192.168.122.1",
      "connected_at": "2022-04-04T19:08:43+0530",
      "encryption": {
        "cipher": "",
        "degree": "none"
      },
      "signing": {
        "cipher": "",
        "degree": "none"
      }
    }
  },
  "locked_files": {
    "//MAINTAINERS": {
      "service_path": "/",
      "filename": "MAINTAINERS",
      "fileid": {
        "devid": 815961915,
        "inode": -61479267,
        "extid": 0
      },
      "num_pending_deletes": 0,
      "opens": [
        {
          "pid": "1:20614",
          "uid": 0,
          "sharemode": {
            "NONE": false,
            "READ": true,
            "WRITE": true,
            "DELETE": false
          },
          "access_mask": {
            "hex": "0x00000003",
            "READ_DATA": true,
            "WRITE_DATA": true,
            "APPEND_DATA": false,
            "READ_EA": false,
            "WRITE_EA": false,
            "EXECUTE": false,
            "READ_ATTRIBUTES": false,
            "WRITE_ATTRIBUTES": false,
            "DELETE_CHILD": false,
            "DELETE": false,
            "READ_CONTROL": false,
            "WRITE_DAC": false,
            "SYNCHRONIZE": false,
            "ACCESS_SYSTEM_SECURITY": false,
            "MAXIMUM_ALLOWED": false,
            "GENERIC_ALL": false,
            "GENERIC_EXECUTE": false,
            "GENERIC_WRITE": false,
            "GENERIC_READ": false,
            "text": "RDWR"
          },
          "oplock": {
            "EXCLUSIVE": false,
            "BATCH": false,
            "LEVEL_II": false,
            "LEASE": false,
            "text": "NONE"
          },
          "lease": {},
          "time": "2022-04-04T19:09:01+0530"
        }
      ]
    }
  }
}
```

**Comment by anoopcs9** [comment]
This is now available with Samba v4.17. See section **JSON support for smbstatus** from [release notes](https://www.samba.org/samba/history/samba-4.17.0.html).
